pages/apis/javascript/latest/js/exn.mdx

```diff
@@ -26,35 +32,44 @@ Provide utilities for dealing with JS exceptions.
 let asJsExn: exn => option(t);
 ```
 
+Returns `Some(exn)` as a JavaScript exception if its argument is an OCaml `Error`, `None` otherwise. *Someone needs to check this; I am not really sure this is an accurate description.*
```

Comment:
~~~suggestion
Returns `Some(jsExn)` if the  input argument `exn` is an actual JS exception, and returns `None` if `exn` is a native Reason exception.

```re example
let throwJsException: unit => 'a = [%raw {|function() { throw "hello"; }|}];

// This will log out "hello", since a JS exception was thrown
try (throwJsException()) {
  | myExn => switch(myExn->Js.Exn.asJsExn) {
     | Some(jsExn) => jsExn->Js.log
     | None => Js.log("this is a Reason exception")
  }
};

exception MyError;

// This will log out "this is a Reason exception", since our raised error is a native Reason exception
try (raise(MyError)) {
  | myExn => switch(myExn->Js.Exn.asJsExn) {
     | Some(jsExn) => jsExn->Js.log
     | None => Js.log("this is a Reason exception")
  }
};
```
~~~

**Example on how to use this (props to @cristianoc for explaining it):**

Reply:
after further discussion, this function seems rather redundant, since the same functionality can be achieved via pattern matching:

```reason

try(throwJsException()) {
  | Js.Exn.Error(_) => Js.log("Js exception");
  | _ => Js.log("Some Reason exception")
}
```

Reply:
How about combining these? Change the explanation to:

> Returns `Some(jsExn)` if the  input argument `exn` is an actual JS exception, and returns `None` if `exn` is a native Reason exception.

and then add:
> Rather than using this function, you can get the same functionality with pattern matching:

with your example.

Reply:
yeah, that sounds good!